pkg/openthread/contrib/platform_radio.c

```diff
@@ -172,20 +182,35 @@ void recv_pkt(otInstance *aInstance, netdev_t *dev)
     otPlatRadioReceiveDone(aInstance, res > 0 ? &sReceiveFrame : NULL, res > 0 ? OT_ERROR_NONE : OT_ERROR_ABORT);
 }
 
+/* create a fake ACK frame: currently does not directly provide a received ACK */
+static inline void _create_fake_ack_frame(bool ackPending)
+{
+    sAckFrame.mPsdu[0] = IEEE802154_FCF_TYPE_ACK;
+    if (ackPending) {
+        sAckFrame.mPsdu[0] |= IEEE802154_FCF_FRAME_PEND;
+    }
+    sAckFrame.mPsdu[1] = 0;
+    sAckFrame.mPsdu[2] = sTransmitFrame.mPsdu[IEEE802154_DSN_OFFSET];
+
+    sAckFrame.mInfo.mRxInfo.mRssi = OT_RADIO_RSSI_INVALID;
+}
+
 /* Called upon TX event */
-void send_pkt(otInstance *aInstance, netdev_t *dev, netdev_event_t event)
+void sent_pkt(otInstance *aInstance, netdev_t *dev, netdev_event_t event)
 {
     (void)dev;
 
     /* Tell OpenThread transmission is done depending on the NETDEV event */
     switch (event) {
         case NETDEV_EVENT_TX_COMPLETE:
             DEBUG("openthread: NETDEV_EVENT_TX_COMPLETE\n");
-            otPlatRadioTxDone(aInstance, &sTransmitFrame, NULL, OT_ERROR_NONE);
+            _create_fake_ack_frame(false);
+            otPlatRadioTxDone(aInstance, &sTransmitFrame, &sAckFrame, OT_ERROR_NONE);
             break;
         case NETDEV_EVENT_TX_COMPLETE_DATA_PENDING:
             DEBUG("openthread: NETDEV_EVENT_TX_COMPLETE_DATA_PENDING\n");
-            otPlatRadioTxDone(aInstance, &sTransmitFrame, NULL, OT_ERROR_NONE);
+            _create_fake_ack_frame(true);
```

Comment:
This seems odd to me. Maybe it is only because of the naming and me not being familiar with openthread. But I would expect ACKs to be created upon receiving a frame, not after sending one.

Reply:
Yes, this "fake" ack is created to report the result of a sent frame to the upper layer. This is not for responding to a received frame.

Reply:
This same technique is used in some of the official platform implementations of Openthread.